test/hotspot/jtreg/vmTestbase/nsk/share/Finalizable.java

```diff
@@ -33,11 +33,18 @@
 public interface Finalizable {
 
     /**
-     * This method will be invoked by <tt>Finalizer</tt> when virtual mashine
+     * This method will be invoked by <tt>Finalizer</tt> when virtual machine
      * shuts down.
      *
      * @throws Throwable if any throwable exception thrown during finalization
      */
     public void finalizeAtExit() throws Throwable;
 
+    /*
+     * This method will register a cleanup method and creates an instance of Finalizer
+     * to register the object for finalization at VM exit.
+     * It is implemented in FinalizableObject.
+     */
+    public void registerCleanup();
```

Comment:
Can you not implement this as a default method?

Reply:
It seems much better to force implementors to provide this method.

Reply:
Why? They all seem to do exactly the same thing.

Reply:
After I moved the `registerCleanup` to the body of a `default` method in the interface, there is no need for the implementors of the `Finalizable` interface to provide this method. All of them can use the default one.

Reply:
> All of them can use the default one

Exactly my point.

Reply:
Default methods for interface classes were invented to solve a problem of compatibility if I remember correctly.  Forcing subclasses to implement the interface method or have a superclass of the subclass to implement the interface method seems like it avoids the problem of silently not registering the cleanup or action that the interface method should force you to do.  To solve the duplicated registerCleanup() cases, the two other classes could extend FinalizableObject then inherit its implementation of registerCleanup().

Reply:
> To solve the duplicated registerCleanup() cases, the two other classes could extend FinalizableObject then inherit its implementation of registerCleanup().

The other two cases already extend the `Log.Logger` and cannot extend `FinalizableObject`.

Reply:
I though default methods were bad design just like default parameters.  maybe not.  I didn’t think yours would compile either but I guess it does.